esmvaltool/preprocessor/__init__.py

```diff
@@ -202,12 +202,16 @@ def preprocess(items, step, **settings):
     if itype.endswith('s'):
         result.append(_run_preproc_function(function, items, settings))
     else:
-        for item in items:
-            result.append(_run_preproc_function(function, item, settings))
+        try:
+            for item in items:
+                result.append(_run_preproc_function(function, item, settings))
+        except TypeError:
+            pass
 
     items = []
     for item in result:
-        if isinstance(item, (PreprocessorFile, Cube, six.string_types)):
+        if isinstance(item,
+                      (PreprocessorFile, Cube, CubeList, six.string_types)):
```

Comment:
I'm not sure if this will work, probably related to your introduction of the `cube_list`?